core-rust/core-api-server/src/core_api/handlers/transaction_status.rs

```diff
@@ -193,6 +193,7 @@ fn map_rejected_payloads_due_to_known_commit(
                 .most_applicable_status()
                 .map(|reason| reason.to_string())
                 .unwrap_or_else(|| {
+                    // TODO(resolve during review): do we ever encounter this?
```

Comment:
_(note from author)_

We should probably discuss it further with @malexandru-rdx .
Maybe this never happens? (so, panic instead?)
Or maybe the status should be `AlreadyCommitted` instead?

(same in LTS endpoint)

Reply:
as discussed on Slack, it seems we can just keep constructing the (definitely known) error in this corner case.
In the latest commit I've migrated the error to its new representation and added some comments, PTAL.